engine/src/main/java/org/camunda/bpm/engine/impl/core/variable/scope/VariableListenerInvocationListener.java

```diff
@@ -51,14 +51,16 @@ protected void handleEvent(VariableEvent event) {
 
     if (sourceScope instanceof ExecutionEntity) {
       addEventToScopeExecution((ExecutionEntity) sourceScope, event);
-    } else if (sourceScope instanceof TaskEntity) {
-      TaskEntity task = (TaskEntity) sourceScope;
-      ExecutionEntity execution = task.getExecution();
-      if (execution != null) {
-        addEventToScopeExecution(execution, event);
-      }
     } else {
-      throw new ProcessEngineException("BPMN execution scope expected");
+      if(sourceScope.getParentVariableScope() instanceof ExecutionEntity) {
```

Comment:
Personally i prefer else if. Maybe we can change this in else if {} else {} instead of if { if () {} else {}} .